nilearn/datasets/func.py

```diff
@@ -1779,49 +1779,112 @@ def fetch_bids_langloc_dataset(data_dir=None, verbose=1):
 
 
 @fill_doc
-def fetch_openneuro_dataset_index(data_dir=None,
-                                  dataset_version='ds000030_R1.0.4',
-                                  verbose=1):
+def fetch_openneuro_dataset_index(
+    data_dir=None,
+    dataset_version='ds000030_R1.0.4',
+    verbose=1,
+):
     """Download a file with OpenNeuro :term:`BIDS` dataset index.
 
+    .. deprecated:: 0.9.2
+        `fetch_openneuro_dataset_index` will be removed in 0.11.
+
     Downloading the index allows to explore the dataset directories
     to select specific files to download. The index is a sorted list of urls.
 
     Parameters
     ----------
     %(data_dir)s
-    dataset_version : string, optional
+    dataset_version : :obj:`str`, optional
         Dataset version name. Assumes it is of the form [name]_[version].
         Default='ds000030_R1.0.4'.
+
+        .. warning:: Any value other than the default will be ignored.
+
+    %(verbose)s'
+
+    Returns
+    -------
+    urls_path : :obj:`str`
+        Path to downloaded dataset index.
+    urls : :obj:`list` of :obj:`str`
+        Sorted list of dataset directories.
+    """
+    warnings.warn(
+        (
+            'The "fetch_openneuro_dataset_index" function was deprecated in '
+            'version 0.9.2, and will be removed in 0.11. '
+            'Please use "fetch_ds000030_urls" instead.'
+        ),
+        DeprecationWarning,
+    )
+
+    DATASET_VERSION = 'ds000030_R1.0.4'
+    if dataset_version != DATASET_VERSION:
+        warnings.warn(
+            (
+                'An improper dataset_version has been provided. '
+                '"ds000030_R1.0.4" will be downloaded.'
+            ),
+            UserWarning,
+        )
+
+    urls_path, urls = fetch_ds000030_urls(data_dir=data_dir, verbose=verbose)
+    return urls_path, urls
+
+
+@fill_doc
+def fetch_ds000030_urls(data_dir=None, verbose=1):
+    """Fetch URLs for files from the ds000030 :term:`BIDS` dataset.
+
```

Comment:
```suggestion

    .. versionadded:: 0.9.2
    
```

Reply:
Thanks for catching that. Should be good now.